### PR TITLE
OSX Travis upgrade xcode to 9.4

### PR DESCRIPTION
**Pull request by hendrikmuhs**
upgrade xcode to 9.4

### CODE REVIEW

File: travis/build_osx_package.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
-set -ex
+set -x
```
Comment (narekgharibyan) [inline travis/build_osx_package.sh]: Don't we want to exit if one of commands in this script returns non-zero exit code ?

**Comment by hendrikmuhs** [comment]
> Great, just curious why `-e` for shell interpreter was removed.

I had massive trouble with this in keyvi-server where I upgraded to xcode 9.4, builds broke with no visible reason until I found out its the `-e`. There are some references to it e.g. https://github.com/travis-ci/docs-travis-ci-com/issues/1672
And in the docs: https://docs.travis-ci.com/user/common-build-problems/#my-build-fails-unexpectedly

(Note that they say it's ok in external scripts, we have separate files, however we `source` them, therefore they are internal although being in an external file)

**Comment by narekgharibyan** [comment]
Hah, I see now. 

I think we `source` them because of these 2 exports:
```
export TMPDIR=/Volumes/ram-disk
export PATH="${HOME}/.pyenv/shims/:/root/.pyenv/bin:${PATH}"
```

Maybe we just need to take them out of these scripts and execute instead of `source` ? 

I'm just worried that if we remove `-e` then we will have builds/tests failures which will not be reported.

**Comment by hendrikmuhs** [comment]
> I'm just worried that if we remove `-e` then we will have builds/tests failures which will not be reported.

Obviously a valid point, osx does not even compile with xcode 9.4 at the moment:
```
In file included from /Users/travis/build/KeyviDev/keyvi/keyvi/bin/keyvi_c/c_api.cpp:29:
/Users/travis/build/KeyviDev/keyvi/keyvi/include/keyvi/dictionary/completion/multiword_completion.h:52:45: error: implicit instantiation of undefined template 'boost::shared_ptr<keyvi::dictionary::Dictionary>'

  explicit MultiWordCompletion(dictionary_t d) : fsa_(d->GetFsa()) {}

                                            ^

/usr/local/include/boost/exception/exception.hpp:15:44: note: template is declared here

namespace boost { template <class T> class shared_ptr; }

                                           ^

In file included from /Users/travis/build/KeyviDev/keyvi/keyvi/bin/keyvi_c/c_api.cpp:30:

/Users/travis/build/KeyviDev/keyvi/keyvi/include/keyvi/dictionary/completion/prefix_completion.h:53:42: error: implicit instantiation of undefined template 'boost::shared_ptr<keyvi::dictionary::Dictionary>'

  explicit PrefixCompletion(dictionary_t d) : fsa_(d->GetFsa()) {}
```

**Comment by narekgharibyan** [comment]
Hah, I believe we can just use `std::shared_ptr` instead, I don't think we have a special reason for using `boost::shared_ptr`, wdyt ?

**Comment by hendrikmuhs** [comment]
> Hah, I believe we can just use `std::shared_ptr` instead, I don't think we have a special reason for using `boost::shared_ptr`, wdyt ?

Fully agree, I somewhat remember that autowrap had problems with `std::shared_ptr` and that's why boost has been used in some places, but I think this is no longer the case.

**Comment by narekgharibyan** [comment]
@hendrikmuhs FYI https://github.com/uweschmitt/autowrap/pull/83

**Comment by hendrikmuhs** [comment]
created #133 and #134 as follop ups

**Comment by narekgharibyan** [comment]
@hendrikmuhs cool, I'll take care of #133

**Pull request merged**